Source/Core/VideoCommon/PixelEngine.h

```diff
@@ -54,14 +65,157 @@ enum class AlphaReadMode : u16
   ReadNone = 2,  // Always read the real alpha value
 };
 
-void Init();
-void DoState(PointerWrap& p);
+// Note: These enums are (assumed to be) identical to the one in BPMemory, but the base type is set
+// to u16 instead of u32 for BitField
+enum class CompareMode : u16
+{
+  Never = 0,
+  Less = 1,
+  Equal = 2,
+  LEqual = 3,
+  Greater = 4,
+  NEqual = 5,
+  GEqual = 6,
+  Always = 7
+};
+
+union UPEZConfReg
+{
+  u16 hex;
+  BitField<0, 1, bool, u16> z_comparator_enable;
+  BitField<1, 3, CompareMode, u16> function;
+  BitField<4, 1, bool, u16> z_update_enable;
+};
+
+enum class SrcBlendFactor : u16
+{
+  Zero = 0,
+  One = 1,
+  DstClr = 2,
+  InvDstClr = 3,
+  SrcAlpha = 4,
+  InvSrcAlpha = 5,
+  DstAlpha = 6,
+  InvDstAlpha = 7
+};
 
-void RegisterMMIO(MMIO::Mapping* mmio, u32 base);
+enum class DstBlendFactor : u16
+{
+  Zero = 0,
+  One = 1,
+  SrcClr = 2,
+  InvSrcClr = 3,
+  SrcAlpha = 4,
+  InvSrcAlpha = 5,
+  DstAlpha = 6,
+  InvDstAlpha = 7
+};
 
-// gfx backend support
-void SetToken(const u16 token, const bool interrupt, int cycle_delay);
-void SetFinish(int cycle_delay);
-AlphaReadMode GetAlphaReadMode();
+enum class LogicOp : u16
+{
+  Clear = 0,
+  And = 1,
+  AndReverse = 2,
+  Copy = 3,
+  AndInverted = 4,
+  NoOp = 5,
+  Xor = 6,
+  Or = 7,
+  Nor = 8,
+  Equiv = 9,
+  Invert = 10,
+  OrReverse = 11,
+  CopyInverted = 12,
+  OrInverted = 13,
+  Nand = 14,
+  Set = 15
+};
+
+union UPEAlphaConfReg
+{
+  u16 hex;
```

Comment:
These unions should probably have `hex` defaulted to zero, otherwise these get constructed uninitialized in the PixelEngine class